### PR TITLE
CPT update

### LINKED ISSUE #4403: Categorical Scientific Colour Maps
**Description of the desired feature**

So I just read Fabio and others' Nature Comms paper at https://doi.org/10.1038/s41467-020-19160-7 and noticed that there are categorical versions of his colour maps (e.g. `batlowS`)! They seem to have been added in version 6, but the PR at #3649 didn't add them into GMT (only the cyclic ones like romaO were added).

[Scientific Categorical Colour Maps](http://www.youtube.com/watch?v=XJcnxmGrxVY "Scientific Categorical Colour Maps")

The relevant page is at http://www.fabiocrameri.ch/categorical.php, and the full list of missing categorical colour maps are:

batlowS, devonS, davosS, osloS,lapazS,actonS,lajollaS,bilbaoS,grayCS,tokyoS,turkuS,ba-makoS, nuukS, hawaiiS, budaS, imolaS

Is it a matter of adding a few more entries to the script at https://github.com/GenericMappingTools/gmt/blob/49669eebc255945f8dc052d70fcb324bde4e7729/admin/build_scientific_colors_cpt.sh to pull these categorical CPTs in from https://doi.org/10.5281/zenodo.4153113, or would it be more work than that?

**Are you willing to help implement and maintain this feature?** Maybe

### PR DESCRIPTION
**Pull request by PaulWessel**
**Description of proposed changes**

This PR refreshes the CPT collection by including the [categorical color maps](http://www.fabiocrameri.ch/categorical.php) from the Scientific Color Map complication.  In the process we rearrange how the CPTs are presented in the documentation.  Closes #4403.

### CODE REVIEW

**Comment by seisman** [comment]
Just a quick comment: now there are no appendix in the cookbook. The names GMT_App_M_xx no longer make sense. Perhaps we can rename them to GMT_CPT_[abcd] or GMT_built-in_CPT_[abcd]?

**Comment by PaulWessel** [comment]
Sure, we could do that, but there i also App E F G J K N O scripts.  Perhaps better to save for a separate PR to do this?

**Comment by seisman** [comment]
> Sure, we could do that, but there is also App E F G J K N O scripts. Perhaps better to save for a separate PR to do this?

OK.

File: admin/build_scientific_colors_cpt.sh [hunk 1]
```diff
@@ -38,51 +39,79 @@ DIR=$1
 VERSION=6.0.4
 cat << EOF > /tmp/cpt.info
 acton|Perceptually uniform sequential colormap, by Fabio Crameri [C=RGB]
+actonS|Perceptually uniform sequential categorical colormap, by Fabio Crameri [C=RGB]
 bamako|Perceptually uniform, low-lightness gradient colormap by Fabio Crameri [C=RGB]
+bamakoS|Perceptually uniform, low-lightness gradient categorical colormap by Fabio Crameri [C=RGB]
 batlow|Perceptually uniform ‘rainbow’ colormap by Fabio Crameri [C=RGB]
+batlowS|Perceptually uniform ‘rainbow’ categorical colormap by Fabio Crameri [C=RGB]
```
Comment (seisman) [inline admin/build_scientific_colors_cpt.sh:46]: Is there any special reason you're using the non-ASCII single quotes `‘` vs `'`?

Comment (PaulWessel) [reply admin/build_scientific_colors_cpt.sh:46]: I am using the C syntax \' in the text.  I think \' and \" gives the right quotes, no?

Comment (seisman) [reply admin/build_scientific_colors_cpt.sh:46]: `'` and `"` are both correct, but the quotes in the following line is non-ASCII. 
```
batlowS|Perceptually uniform ‘rainbow’ categorical colormap by Fabio Crameri [C=RGB]
```

Comment (PaulWessel) [reply admin/build_scientific_colors_cpt.sh:46]: Sorry formatting in the comment.  Per C manual I can use

`\' and \" for quotes`

which is what I do. I am not using` ‘ `

Comment (seisman) [reply admin/build_scientific_colors_cpt.sh:46]: I'm talking about the `admin/build_scientific_colors_cpt.sh` bash script and the generated CPTs. 

If you go to the share/cpt directory, and run `file *.cpt`, you will see while most CPTs are ASCII plaintext files, some are not, due to the non-ASCII `‘`.
```
abyss.cpt:       ASCII text
acton.cpt:       ASCII text
bamako.cpt:      ASCII text
bathy.cpt:       ASCII text
batlow.cpt:      UTF-8 Unicode text
berlin.cpt:      ASCII text
bilbao.cpt:      ASCII text
broc.cpt:        ASCII text
```

Comment (PaulWessel) [reply admin/build_scientific_colors_cpt.sh:46]: Yes, sorry, I was looking in gmt_cpt_masters.h....
I see what you mean and will fix, and also rename the build script to replace those underscores with hyphens, like our other build scripts.

**Comment by PaulWessel** [comment]
BTW, any quickway of tracking down the offending lines here:
./gmtregress.c: c program text, UTF-8 Unicode text
./gmt_init.c: c program text, UTF-8 Unicode text

**Comment by seisman** [comment]
> BTW, any quickway of tracking down the offending lines here:
> ./gmtregress.c: c program text, UTF-8 Unicode text
> ./gmt_init.c: c program text, UTF-8 Unicode text

https://pages.cs.wisc.edu/~markm/ascii.html Try copy and paste the source code to this online tool

**Comment by PaulWessel** [comment]
Thanks, forgot about references copied from EndNote...

**Comment by PaulWessel** [comment]
I have a bug in preparing the CSM categorical as colors 2 and 3 are identical.  Will look and then update.

**Comment by PaulWessel** [comment]
I think I have fixed remaining issues with the CPTs - please have a look, @seisman.

File: admin/build-scientific-colors-cpt.sh [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+	else
+		cptdir=${cpt}
+	fi
+	cat <<- EOF >> gmt_cpts/$cpt.cpt
+	#
+	#	www.fabiocrameri.ch/visualisation
+	#
+	# License: Creative Commons Attribution 4.0 International License
+	# Copyright (c) 2020, Fabio Crameri All rights reserved.
+	# Crameri, F., (2020). Scientific colour-maps. Zenodo. https://zenodo.org/record/4153113
```
Comment (seisman) [inline admin/build-scientific-colors-cpt.sh:114]:
```suggestion
	# Crameri, F., (2020). Scientific colour maps. Zenodo. https://zenodo.org/record/4153113
```

Comment (seisman) [reply admin/build-scientific-colors-cpt.sh:114]: You also need to update all CPTs after making the above change.

**Comment by seisman** [comment]
**Old:**
image

**New:**
image

The vertical separations between rows are a little too small in the new layout.

**Comment by PaulWessel** [comment]
OK, well I changed it from 0.75 to 0.6 and then 0.5 which I thought was OK, but I can go back to 0.6 if you think that will look better.  0.75 was too much I think.  Try 0.6?

**Comment by seisman** [comment]
image

I think 0.6 is good.

**Comment by PaulWessel** [comment]
0.6i implemented

File: admin/build-scientific-colors-cpt.sh (hunk 2 above)
Comment (seisman) [reply admin/build-scientific-colors-cpt.sh:114]: Don't forget to commit this change and update all CPTs.

**Comment by PaulWessel** [comment]
OK too many zooms, now should be OK.

**Review by seisman — APPROVED** [review]

**Pull request merged**